test/vmapi.test.js

```diff
@@ -110,6 +125,116 @@ function waitForValue(fn, params, prop, value, callback) {
     return check();
 }
 
+function connectToMoray(callback) {
+    var MORAY_CLIENT_CONFIG = {
+        host: process.env.MORAY_IP || '10.99.99.17',
+        port: 2020,
+        log: new Logger({
+            name: 'moray',
+            level: 'info',
+            serializers: restify.bunyan.serializers
+        })
+    };
+    var morayClient = moray.createClient(MORAY_CLIENT_CONFIG);
+
+    morayClient.on('connect', function onConnect() {
+        return callback(null, morayClient);
+    });
+}
+
+/*
+ * Creates "nbTestVms" VMs in moray. These test VMs will be created with an
+ * alias of TEST_VMS_ALIAS to be able to differentiate test VMs from non-test
+ * VMs when, for instance, it's time to clean them up.
+ * "vmProperties" is an object that contains properties names and properties
+ * values that need to be set for all created VMs.
+ * "callback" is called with an error object as its first argument, which is
+ * null or undefined if there was no error.
+ */
+function createTestVms(nbTestVms, vmProperties, callback) {
+    assert.number(nbTestVms, 'nbTestVms');
+    assert.object(vmProperties, 'vmProperties');
+    assert.func(callback, 'callback');
+
+    vasync.waterfall([
+        connectToMoray,
+        function createVms(morayClient, next) {
+            var addVmsQueue = vasync.queue(createTestVm, 10);
+            for (var i = 0; i < nbTestVms; ++i) {
+                addVmsQueue.push(libuuid.create());
+            }
+
+            addVmsQueue.close();
+            addVmsQueue.on('end', function testVmsCreationDone() {
+                return next(null, morayClient);
+            });
+
+            function createTestVm(vmUuid, callback) {
+                var vmObject = {
```

Comment:
This is re-using the 'callback' var. 
I'm surprised this is passing 'make check'.